xarray/backends/zarr.py

```diff
@@ -0,0 +1,474 @@
+        # TODO: possibly wrap zarr array in dask with aligned chunks
+        data = indexing.LazilyIndexedArray(ZarrArrayWrapper(name, self))
+        dimensions, attributes = _get_zarr_dims_and_attrs(
+                                    zarr_array, self._DIMENSION_KEY)
+        encoding = {'chunks': zarr_array.chunks,
+                    'compressor': zarr_array.compressor,
+                    'filters': zarr_array.filters,
+                    'fill_value': zarr_array.fill_value}
+
+        var = Variable(dimensions, data, attributes, encoding)
+
+        if self._auto_chunk:
+            from dask.base import tokenize
+            # is this token enough?
+            token = tokenize(zarr_array)
+            name = 'zarr_array-%s' % token
+            # do we need to worry about the zarr synchronizer / dask lock?
+            lock = self._synchronizer
+            print("Chunking variable")
+            var = var.chunk(chunks=zarr_array.chunks, name=name, lock=lock)
+
+        return var
+
+
+    def get_variables(self):
+        with self.ensure_open(autoclose=False):
+            return FrozenOrderedDict((k, self.open_store_variable(k, v))
+                                     for k, v in self.ds.arrays())
+
+    def get_attrs(self):
+        with self.ensure_open(autoclose=True):
+            _, attributes = _get_zarr_dims_and_attrs(self.ds,
+                                                     self._DIMENSION_KEY)
+            attrs = FrozenOrderedDict(attributes)
+            return attrs
+
+    def get_dimensions(self):
+        with self.ensure_open(autoclose=True):
+            dimensions, _ = _get_zarr_dims_and_attrs(self.ds,
+                                                     self._DIMENSION_KEY)
+            return dimensions
+
+    def set_dimension(self, name, length):
+        with self.ensure_open(autoclose=False):
+            self.ds.attrs[self._DIMENSION_KEY][name] = length
+
+    def set_attribute(self, key, value):
+        with self.ensure_open(autoclose=False):
+            _, attributes = _get_zarr_dims_and_attrs(self.ds,
+                                self._DIMENSION_KEY)
+            attributes[key] = value
+
+    def prepare_variable(self, name, variable, check_encoding=False,
+                         unlimited_dims=None):
+
+        attrs = variable.attrs.copy()
+        dims = variable.dims
+        dtype = variable.dtype
+        shape = variable.shape
+
+        # TODO: figure out how zarr should deal with unlimited dimensions
+        self.set_necessary_dimensions(variable, unlimited_dims=unlimited_dims)
+
+        # netcdf uses pop not get...yet it works. Why?
+        # here we are basically duplicating zarr's own internal fill_value
+        # in an attribute. This seems redundant and error prone. How can
+        # we do better?
+        fill_value = attrs.get('_FillValue', None)
+        if fill_value in ['\x00']:
+            fill_value = None
```

Comment:
As I explain in my comment, I am a bit confused about how to handle `fill_value`. zarr has a built in fill_value mechanisms, and I don't know the best way to plug it into xarray.